### PR TITLE
Microsoft.PolicyInsights/checkPolicyRestrictions API

### PR DESCRIPTION
**Pull request by pilor**
Add the checkPolicyRestrictions API to Microsoft.PolicyInsights.  This is a pure copy/paste (other than changing api-version from preview to stable) from the private repo PR here:  https://github.com/Azure/azure-rest-api-specs-pr/pull/1707

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 1]
```diff
@@ -1398,6 +1398,11 @@
           "description": "Expression evaluated.",
           "type": "string"
         },
+        "expressionKind": {
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json:1401]: This was an oversight in the previous swagger spec.  The readonly property has been part of the API response from the time the api-version was introduced.

**Comment by pilor** [comment]
Migrating the ArmSignedOff label from the private repo as no changes were made to the API between [that PR](https://github.com/Azure/azure-rest-api-specs-pr/pull/1707) and this.  Only the api-version was changed to be non-preview.

**Comment by pilor** [comment]
The .NET SDK failure is due to the changes from this previous PR not being present in the SDK repo yet:  https://github.com/Azure/azure-rest-api-specs/pull/7909

I'm following up with the author of that change to ensure they get those changes there soon.

**Comment by jianyexi** [comment]
> The .NET SDK failure is due to the changes from this previous PR not being present in the SDK repo yet: #7909
> 
> The python breaking change is the same issue. The breaking change was in the above PR and hasn't been generated in python yet...
> 
> I'm following up with the author of that change to ensure they get those changes there soon.

Just want to confirm if  this work has done

**Comment by pilor** [comment]
> > The .NET SDK failure is due to the changes from this previous PR not being present in the SDK repo yet: #7909
> > The python breaking change is the same issue. The breaking change was in the above PR and hasn't been generated in python yet...
> > I'm following up with the author of that change to ensure they get those changes there soon.
> 
> Just want to confirm if this work has done

Someone on my team is working on it at the moment but its not done yet.

**Comment by jianyexi** [comment]
> > > The .NET SDK failure is due to the changes from this previous PR not being present in the SDK repo yet: #7909
> > > The python breaking change is the same issue. The breaking change was in the above PR and hasn't been generated in python yet...
> > > I'm following up with the author of that change to ensure they get those changes there soon.
> > 
> > 
> > Just want to confirm if this work has done
> 
> Someone on my team is working on it at the moment but its not done yet.

Any update?

**Comment by jianyexi** [comment]
@pilor  There is Python SDK breaking change , could you confirm its impact , or do you really want to bump a SDK version ?

**Comment by pilor** [comment]
> @pilor There is Python SDK breaking change , could you confirm its impact , or do you really want to bump a SDK version ?

The python SDK breaking change was from a previous PR linked above (#7909) that hasn't been generated in the python repo yet.  The people that made that change are in the process of getting it out to the various SDKs.

**Comment by jianyexi** [comment]
> > @pilor There is Python SDK breaking change , could you confirm its impact , or do you really want to bump a SDK version ?
> 
> The python SDK breaking change was from a previous PR linked above (#7909) that hasn't been generated in the python repo yet. The people that made that change are in the process of getting it out to the various SDKs.

Thanks, pls let know if the PR is ready to merge

**Comment by pilor** [comment]
@jianyexi The SDK breaking change should finally be resolved as the .NET SDK has been updated to reflect the previous breaking change.  Just merged from master to trigger a new validation pass to verify.  Python SDK generation request is underway which should resolve the Python breaking change warning.  If you don't mind I'd prefer to wait for the language specific SDKs to be generated off of the existing tag in "master" and then verify the python breaking change goes away.

If you feel this has been open for too long I can close it and re-open when python generation is done.

**Comment by pilor** [comment]
@jianyexi This should be good to go now.  Is there a way to re-trigger the SDK validation against python to ensure the breaking change warning goes away?

**Comment by pilor** [comment]
@jianyexi Pyrhon/.NET are good to go now.  However, it looks like the generated code form some paginated APIs that aren't touched in this PR changed in the GO sdk.  The funcs now take the current page as a param which they did not take before...  Is this a known issue in GO generation right now?

**Comment by jianyexi** [comment]
> @jianyexi Pyrhon/.NET are good to go now. However, it looks like the generated code form some paginated APIs that aren't touched in this PR changed in the GO sdk. The funcs now take the current page as a param which they did not take before... Is this a known issue in GO generation right now?

@ArcturusZhang could you take a look ?

**Review by ArcturusZhang — APPROVED** [review]
Breaking change and failure in go SDK generation are all expected and will be resolved soon

File: specification/policyinsights/resource-manager/readme.md [hunk 2]
```diff
@@ -66,8 +66,37 @@ directive:
...
+  - suppress: PostOperationIdContainsUrlVerb
+    reason: The operation can be performed at multiple scopes. The operationId needs to indicate the scope.
+    where:
+      - $.paths["/subscriptions/{subscriptionId}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation"].post.operationId
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation"].post.operationId
+      - $.paths["/subscriptions/{subscriptionId}/providers/Microsoft.PolicyInsights/checkPolicyRestrictions"].post.operationId
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PolicyInsights/checkPolicyRestrictions"].post.operationId
+
+  - suppress: ListInOperationName
+    reason: The result is an object that contains two arrays, it is not a top level list nor what would be considered a list operation.
```
[the remark this answers is not in the document]
Comment (pilor) [reply specification/policyinsights/resource-manager/readme.md]: Removed

**Comment by pilor** [comment]
@jianyexi ready to merge unless you have additional feedback

**Review by jianyexi — APPROVED** [review]

**Pull request merged**